main.py

```diff
@@ -36,21 +64,32 @@ def api_text_to_speech():
     f.write(data)
     f.close()
 
-    result = model.transcribe(filename)
+    max_attempts = 5
+    for attempt in range(max_attempts):
+        result = model.transcribe(filename, initial_prompt=' '.join(hint_phrases))
+        processed_text = clean_text(result['text'])
+        if processed_text:
+            break
 
     os.remove(filename)
 
     print(result)
     cleaned_text = clean_text(result['text'])
+    
+    # Post-processing: Check if the transcribed text is in hint_phrases
+    if cleaned_text not in hint_phrases:
+        cleaned_text = ''
```

Comment:
This should be an option as well. I assume some people would want the result to be sent even if it's not matching exactly the hint phrases.

Reply:
I added
`parser.add_argument('--strict-matching', type=bool, help="Only return text if it's found in hint phrases", default=False)`
and modified the referenced lines like so:
```
# Post-processing: Check if the transcribed text is in hint_phrases
    if args.strict_matching and cleaned_text not in hint_phrases:
        cleaned_text = ''

    print("Clean text:")
    print(cleaned_text)
```

Now, the container can be started like this:
`docker run -it --gpus all -p 4444:4444 whisper-rhasspy-api --filter-chars ".,-?'\!\":;<>[]{}()" --whisper-model large`

This ensures that without `--strict-matching True`, the script will always return what Whisper understood. Passing this argument will require the user to use `-v /path/to/host/file:/path/to/container/file` and `--hint-phrases-path /path/to/container/file` as well.

Reply:
@Zeevai again I don't see these changes.

Could you also update the "documentation" in the readme on these arguments and ensure that if strict-matching is set to True hint_phrases have to be specified, if they are not throw an error on start up.